File: IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgram3DFisheye4PFactory.m
//
//  IRGLProgram3DFisheye4PFactory.m
//  IRPlayer
//
//  Created by Phil on 2019/7/5.
//  Copyright © 2019 Phil. All rights reserved.
//

#import "IRGLProgram3DFisheye4PFactory.h"
#import <IRPlayer_swift/IRPlayer_swift-Swift.h>

@implementation IRGLProgram3DFisheye4PFactory

-(IRGLProgram2D*)createIRGLProgramWithPixelFormat:(IRPixelFormat)pixelFormat withViewprotRange:(CGRect)viewprotRange withParameter:(IRMediaParameter*)parameter{
    return [IRGLProgramFactory createIRGLProgram3DFisheye4PWithPixelFormat:pixelFormat viewportRange:viewprotRange parameter:(IRMediaParameter*)parameter];
}

@end

File: Sources/IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgram3DFisheye4PFactory.swift
//
//  IRGLProgram3DFisheye4PFactory.swift
//  IRPlayer-swift
//
//  Created by irons on 2024/9/13.
//

import Foundation

class IRGLProgram3DFisheye4PFactory: IRGLProgram2DFactory {

    override func createIRGLProgram(pixelFormat: IRPixelFormat, viewportRange: CGRect, parameter: IRMediaParameter?) -> IRGLProgram2D {
        return IRGLProgramFactory.createIRGLProgram3DFisheye4P(pixelFormat: pixelFormat, viewportRange: viewportRange, parameter: parameter)!
    }
}
